### PR TITLE
Pandas Profiling v2.0.0

### PR DESCRIPTION
**Pull request by sbrugman**
With 14 commits, 115 files changes and 20+ issues resolved, Pandas Profiling v2.0.0 will be a big leap forward. In addition to the [continuous integration tests](https://travis-ci.com/pandas-profiling/pandas-profiling), any help in reviewing the code are highly appreciated!

### CODE REVIEW

File: pandas_profiling/__init__.py [hunk 1]
```diff
@@ -1,153 +1,153 @@
-# -*- coding: utf-8 -*-
 """Main module of pandas-profiling.
 
-Docstring is compliant with NumPy/SciPy documentation standard:
-https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt
-See also for a short description of docstring:
-https://stackoverflow.com/questions/3898572/what-is-the-standard-python-docstring-format
+.. include:: ../README.md
```
Comment (eyaltra) [inline pandas_profiling/__init__.py]: what does this means ?:) is this on purpuse

Comment (sbrugman) [reply pandas_profiling/__init__.py]: This is markup (restructuredtext) to include the README in the documentation!

File: pandas_profiling/__init__.py [hunk 2]
```diff
@@ -1,153 +1,153 @@
...
-                outputfile = 'profile_' + str(hash(self)) + ".html"
-            # TODO: should be done in the template
-            with codecs.open(outputfile, 'w+b', encoding='utf8') as self.file:
-                self.file.write(templates.template('wrapper').render(content=self.html))
-
-    def to_html(self):
+        if type(output_file) == str:
+            output_file = Path(output_file)
+
+        with output_file.open("w", encoding="utf8") as f:
```
Comment (eyaltra) [inline pandas_profiling/__init__.py]: maybe allow encoding as parameter with a default of utf8

File: pandas_profiling/model/base.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        True if the variable is of type datetime.
+    """
+    is_date_value = pd.api.types.is_datetime64_dtype(series)
+    try:
+        series.apply(_date_parser)
+        warnings.warn(
+            'Column "{}" appears to be containing only date/datetime values. You might consider '
+            "changing the type to datetime (pd.to_datetime())".format(series.name)
+        )
+    except ValueError:
```
Comment (eyaltra) [inline pandas_profiling/model/base.py]: cleaner way will be with ignore() context manager https://docs.python.org/3/library/contextlib.html

File: pandas_profiling/model/correlations.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+    Args:
+        variables: A dict with column names and variable types.
+        df: The DataFrame with variables.
+
+    Returns:
+        A dictionary containing the correlation matrices for each of the active correlation measures.
+    """
+    correlations = {}
+    if config["correlations"]["pearson"].get(bool):
```
Comment (eyaltra) [inline pandas_profiling/model/correlations.py]: we can do the first 3 correlation with :
corr_types = [pearson, spearman, kendall]
for corr_type in corr_types:
        correlation = df.corr(method=corr_type)
        if len(correlation) > 0:
            correlations[corr_type] = correlation

**Comment by eyaltra** [comment]
@sbrugman i would love to help, what can i do?

File: pandas_profiling/__init__.py (hunk 1 above)
Comment (eyaltra) [reply pandas_profiling/__init__.py]: cool never knew you can inject markdown via .. include: prefix

**Comment by sbrugman** [comment]
> @sbrugman i would love to help, what can i do?

Hi @eyaltra. Great review suggestions, thank you!
You could start to see if there is anything that interests you here:
https://github.com/pandas-profiling/pandas-profiling/issues

I am working on a roadmap, so if you can't find anything in the issues, we could discuss features in the following areas: data analysis, performance (parallel processing), interactivity, user experience, machine learning profiling and security/restricted environment compatibility.

File: pandas_profiling/model/base.py (hunk 3 above)
Comment (sbrugman) [reply pandas_profiling/model/base.py]: Elegant, will change this.

File: pandas_profiling/model/base.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        try:
+            result = series[~series.isnull()].astype(str).apply(urlparse)
+            return result.apply(lambda x: all([x.scheme, x.netloc, x.path])).all()
+        except ValueError:
+            return False
+    else:
+        return False
+
+
+# def is_path(x):
```
Comment (eyaltra) [inline pandas_profiling/model/base.py]: dead code:)

**Pull request merged**